sql/core/src/main/scala/org/apache/spark/sql/catalyst/analysis/ResolveSessionCatalog.scala

```diff
@@ -475,47 +475,45 @@ class ResolveSessionCatalog(
         propertyKey)
 
     case DescribeFunctionStatement(CatalogAndIdentifier(catalog, functionIdent), extended) =>
-      val functionIdentifier = if (isSessionCatalog(catalog)) {
-        functionIdent.asMultipartIdentifier match {
-          case Seq(db, fn) => FunctionIdentifier(fn, Some(db))
-          case Seq(fn) => FunctionIdentifier(fn, None)
-          case _ =>
-            throw new AnalysisException(s"Unsupported function name '${functionIdent.quoted}'")
-        }
-      } else {
-        throw new AnalysisException ("DESCRIBE FUNCTION is only supported in v1 catalog")
-      }
-      DescribeFunctionCommand(functionIdentifier, extended)
+      val (database, function) =
+        parseSessionCatalogFunctionIdentifier("DESCRIBE FUNCTION", catalog, functionIdent)
```

Comment:
Just a question. Is it hard to make `SessionCatalogAndFunction`?

Reply:
It would be possible in the case that we would have v2 catalog functions to resolve it. But for now we need to handle an specific error for this case

`throw new AnalysisException(s"$sql is only supported in v1 catalog")`

If we don't handle it here, we find and unresolved exception that is not clear.
It is the same situation that we have with the commands that use parseV1Table function.

Reply:
I'm not sure what are you proposing exactly. We could have something like that:
  
```
case DescribeFunctionStatement(SessionCatalogAndFunction(catalog, functionIdent), extended) =>
...
case DropFunctionStatement(SessionCatalogAndFunction(catalog, functionIdent), ifExists, isTemp) =>
...
case CreateFunctionStatement(SessionCatalogAndFunction(catalog, functionIdent),
      className, resources, isTemp, ignoreIfExists, replace) =>
...
```
where SessionCatalogAndFunction is an extractor that tests if isSessionCatalog, something like SessionCatalogAndTable do.
In the case that !isSessionCatalog it implies that any of this cases match and the result is 
`unresolved operator 'CreateFunctionStatement [testcat, ns1, ns2, fun]` for Create function 

How could we catch this case generically and throw a generic exception like you propose?

Reply:
Ah OK. This requires throwing an exception inside the extractor, which may not be ideal.